File: 10.cpp
#define ANSI_COLOR_RED "\x1b[31m"
#define ANSI_COLOR_RESET "\x1b[0m"
#define ANSI_COLOR_GREEN "\x1b[32m"
#include <iostream>
#include <limits>
int main() {
	int year, i = 0, j = 1;
	std::cout << ANSI_COLOR_RED;
	std::cout << "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~" << std::endl;
	std::cout << "~\t\t\tLEAP YEAR TEST\t\t\t~" << std::endl;
	std::cout << "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~" << std::endl << std::endl;
	std::cout << ANSI_COLOR_RESET;
	std::cout << "ENTER YEAR: ";
	std::cin >> year;
	while (std::cin.fail()) {
		std::cin.clear();
		std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
		std::cout << "\nBad entry. Enter a YEAR: ";
		std::cin >> year;
	}
	if (year % 400 == 0) std::cout << year << " is a leap year.";
	else if (year % 100 == 0) std::cout << year << " is not a leap year.";
	else if (year % 4 == 0) std::cout << year << " is a leap year.";
	else { std::cout << year << " is not a leap year."; }
	std::cout << std::endl << std::endl << "\nNEXT 20 LEAP YEARS FROM INPUT YEAR: ";
	while (i < 20) {
		if ((year + j) % 400 == 0 || (year + j) % 4 == 0) {
			if (i == 0) std::cout << "\t" << year + j << std::endl;
			else { std::cout << "\t\t\t\t\t" << year + j << std::endl; }
			i++;
		}
		j++;
	}
	std::cout << ANSI_COLOR_GREEN << "\n\n" << std::endl;
	return 0;
}

File: 10.cs
void yearCheck(int y)
{
    if (y % 400 == 0) Console.Write(y + " is a leap year.");
    else if (y % 100 == 0) Console.Write(y + " is not a leap year.");
    else if (y % 4 == 0) Console.Write(y + " is a leap year.");
    else
    {
        Console.Write(y + " is not a leap year.");
    }
}

void next20LeapYears(int y)
{
    int i = 0, j = 1;
    
    while (i < 20)
    {
        if ((y + j) % 400 == 0 || (y + j ) % 4 == 0)
        {
            if (i == 0) Console.WriteLine("\t" + y + j);
            else
            {
                Console.WriteLine("\t\t\t\t\t" + y + j);
            }
            i++;
        }
        j++;
    }
}

Console.ForegroundColor = ConsoleColor.Red;
Console.WriteLine("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");
Console.WriteLine("~\t\t\tLEAP YEAR TEST\t\t\t~");
Console.WriteLine("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n");
Console.ForegroundColor = ConsoleColor.White;

try
{
    Console.Write("ENTER YEAR: ");
    string year = Console.ReadLine();
    int varYear = int.Parse(year);
    yearCheck(varYear);
    Console.Write("\n\n\nNEXT 20 LEAP YEARS FROM INPUT YEAR: ");
    next20LeapYears(varYear);
} 
catch (FormatException fE) 
{
    Console.WriteLine("\n" + fE.Message);
}

Console.ForegroundColor = ConsoleColor.Green;
Console.WriteLine("\n\n");

return 0;
